Fix: Add Account fails when full server uri is used

sockenklaus opened the pull request:
Implemented a function that extracts the host part from a URL,  removes the protocol and anything that may come after a :, a / or a  ?.

Closes #739

dessalines commented on app/src/main/java/com/jerboa/Utils.kt:
```diff
@@ -1231,6 +1231,18 @@ fun scrollToPreviousParentComment(
     }
 }
 
+/**
```
Someone should really make a library for this, if its functionality isn't already included in URL.

MV-GH replied:
But this is the java ecosystem, of course something like this exists already.

```kt
Patterns.WEB_URL.matcher(url).matches() // To check if a valid URL
println(URL("https://lemmy.ml").host) // lemmy.ml
```

dessalines replied:
Perfect

dessalines reviewed (approved):
LGTM, although I really wish we didn't have to use regex for something like this, and could use a library.

twizmwazin reviewed (changes requested):
Let's use @MV-GH's solution

sockenklaus commented:
> Let's use @MV-GH's solution

But regex is fun! 😢

No, seriously: @MV-GH's solution is clearly more elegant. I will adjust the PR later the day.

twizmwazin commented:
In some cases regex is being used because the links are metaverse links that aren't actually valid URLs, or at least I don't think they are valid. `@user@instance`, `!community@instance` mainly. If it isn't one of those I agree using regex is sub-optimal and fixing those would be welcome.

sockenklaus commented:
I reworked the function but didn't ended up using `Patterns.WEB_URL` to match against because for some reason it doesn't handle hosts without domain-part (i.e. `localhost`). I think local hosts are useful for local testing.

Also i added [Roboelectric](https://robolectric.org/) to the project because i didn't see another way to test my function without mocking. But I'm fairly new to using and writing unit tests to correct me if i was wrong using that package.

MV-GH commented:
Could you not use `@RunWith(JUnitParamsRunner::class)` ?
see `com/jerboa/ui/components/common/LemmyLinkPluginTest.kt`
